src/lib.rs

```diff
@@ -126,12 +126,74 @@ pub struct SelfEncryptor<'a> {
 }
 
 impl<'a> SelfEncryptor<'a> {
-  /// This is the only constructor, for encryptor object
+  /// This is the only constructor, for encryptor object.
   /// Each SelfEncryptor is used for a single file.
   /// The parameters are a DataMap and Storage.
-  /// If new file use DataMap::None as first parameter
+  /// If new file, use DataMap::None as first parameter.
   /// The get and put of Storage need to be implemented to
   /// allow the SelfEncryptor to store encrypted chunks and retrieve when necessary.
+  // # Example usage
+  // ```rust
+  // let mut se = SelfEncryptor::new(&mut my_storage as MyStorage, datamap::DataMap::None);
+  // ```
+  //
+
+/// # Example
+///
+/// SelfEncryptor Example
+///
+/// ```rust
+/// extern crate self_encryption
+/// extern crate rand;
+/// use use self_encryption::*;
+///
+/// fn main() {
```

Comment:
Again should this not take some user input to encrypt a file on disk or similar and then also to decrypt it. (see docopts lib for a nice interface).